test/gtest/Makefile.am

```diff
@@ -107,6 +104,11 @@ gtest_SOURCES = \
 	ucs/test_twheel.cc \
 	ucs/test_frag_list.cc
 
+if ! ENABLE_SYMBOL_OVERRIDE
+gtest_SOURCES += \
+	ucm/malloc_hook.cc
```

Comment:
Why to get into this complexity ? This is much easier solution for relatively exotic case. I don't think that we have to rewrite everything just for this case.